inc/class-post-type.php

```diff
@@ -21,6 +21,10 @@ public static function get_accessible_post_types() {
 		$post_types = get_post_types( array( 'public' => true ) );
 		$post_types = array_filter( $post_types, 'is_post_type_viewable' );
 
+		if ( WPSEO_Options::get( 'disable-attachment' ) ) {
+			$post_types = self::filter_attachment_post_type( $post_types );
```

Comment:
To prevent handling edgecase in this class and to keep this as clean as possible I would move this statement outside this class. Perhaps somewhere in `WPSEO_Admin`.

```php
if ( WPSEO_Options::get( 'disable-attachment' ) ) {
     add_filter( 'wpseo_accessible_post_types', array( 'WPSEO_Post_Type', 'filter_attachment_post_type' ) );
}

Reply:
:+1: I'll try to add new phpunit test which covers situation when media redirect is enabled/disabled.

Reply:
Awesome 👍 I will test and merge after the change :)